### PR TITLE
Replace .getRecentEvents with .getRecentEventsSortedByBlockNumber for related events for each card. #3955

### LINKED ISSUE #3955: Replace .getRecentEvents with .getRecentEventsSortedByBlockNumber for related events for each card.
getRecentEvents() returns only 100 events, that could cause error with creating activities (missing events)
replace with fetching only filtered event instances,

### PR DESCRIPTION
**Pull request by oa-s**
Closes #3955

### CODE REVIEW

**Comment by oa-s** [comment]
### Code execution stats:
**After PR**
XXX: Took 0.02313709259033203 seconds for contractsAndCards: 408
XXX: Took 0.02388596534729004 seconds for contractsAndCards: 408
XXX: Took 0.05323898792266846 seconds for contractsAndCards: 408
XXX: Took 0.12642109394073486 seconds for contractsAndCards: 984
XXX: Took 0.06773591041564941 seconds for contractsAndCards: 984

**Before PR**
XXX: Took 0.42520391941070557 seconds for contractsAndCards: 984
XXX: Took 0.020550966262817383 seconds for contractsAndCards: 408
XXX: Took 0.07148492336273193 seconds for contractsAndCards: 984
XXX: Took 0.006186962127685547 seconds for contractsAndCards: 408
XXX: Took 0.07352697849273682 seconds for contractsAndCards: 984

**Statistics after loading `victor928.eth` after delating the app**
XXX: Took 5.805492401123047e-05 seconds for contractsAndCards: 0
XXX: Took 0.04068708419799805 seconds for contractsAndCards: 792
XXX: Took 0.04105401039123535 seconds for contractsAndCards: 804
XXX: Took 0.04412400722503662 seconds for contractsAndCards: 840
XXX: Took 0.042948007583618164 seconds for contractsAndCards: 840

File: AlphaWallet/TokenScriptClient/Models/EventsActivityDataStore.swift [hunk 1]
```diff
@@ -48,6 +48,15 @@ class EventsActivityDataStore: EventsActivityDataStoreProtocol {
         cachedRecentEventsSubscribable[subscription] = nil
     }
 
+    func getRecentEventsSortedByBlockNumber(forContract contract: AlphaWallet.Address, server: RPCServer, eventName: String, interpolatedFilter: String) -> Results<EventActivity> {
```
Comment (hboon) [inline AlphaWallet/TokenScriptClient/Models/EventsActivityDataStore.swift:51]: Should we add a function comment that sorting by date implies sorting by block number?

**Review by hboon — APPROVED** [review]

File: AlphaWallet/TokenScriptClient/Models/EventsActivityDataStore.swift (hunk 1 above)
Comment (oa-s) [reply AlphaWallet/TokenScriptClient/Models/EventsActivityDataStore.swift:51]: in this case we can update it with replacing `.sorted(byKeyPath: "date", ascending: false)` with `.sorted(byKeyPath: "blockNumber")`

Comment (hboon) [reply AlphaWallet/TokenScriptClient/Models/EventsActivityDataStore.swift:51]: Oh yeah, probably faster too. Remember `ascending: false`.

**Pull request merged**